tests/local/cheyenne/model_test.sh

```diff
@@ -0,0 +1,197 @@
+domain_dir=/glade/work/jamesmcc/domains/private/CONUS
+
+# Getops
+TEMP=\
+`getopt \
+    -o h,c:,r: \
+    --long help,compiler:,mpi:,config:,ncores:,nnodes:,queue:,account:,walltime:,reference_update:,domain_dir: \
+    -n 'Model Testing' -- "$@"`
+
+if [ $? != 0 ] ; then echo "Terminating..." >&2 ; exit 1 ; fi
+# Note the quotes around `$TEMP': they are essential!
+eval set -- "$TEMP"
+while true; do
+    case "$1" in
+        -h | --help ) echo "$the_help"; exit 0 ;;
+        -c ) candidate_dir="$2"; shift 2;;
+        -r ) reference_dir="$2"; shift 2;;
+        --reference_update ) reference_update="$2"; shift 2;;
+        --compiler ) compiler="$2"; shift 2;;
+        --mpi ) mpi="$2"; shift 2;;
+        --config ) config="$2"; shift 2;;
+        --ncores ) ncores="$2"; shift 2;;
+        --nnodes ) nnodes="$2"; shift 2;;
+        --queue ) queue="$2"; shift 2;;
+        --account ) account="$2"; shift 2;;
+        --walltime ) walltime="$2"; shift 2;;
+        --domain_dir ) domain_dir="$2"; shift 2;;
+        -- ) shift; break ;;
+        * ) break ;;
+    esac
+done
+
+# Required variables.
+if [ -z $candidate_dir ] | [ -z $reference_dir ]; then
+    echo "The candidate_dir and reference_dir arguments are both required but not found. Exiting."
+    exit 1
+fi
+
+#-------------------------------------------------------
+# Modules.
+# Default intel and gnu compiler versions if the generics are passed
+if [ "$compiler" == ifort ]; then
+    compiler_module=intel/17.0.1
+elif [ "$compiler" == gfort ]; then
+    compiler_module=gnu/7.0.1
+else
+    compiler_module=$compiler
+    if [[ "$compiler" == *intel* ]]; then
+        compiler=ifort
+    elif [[ "$compiler" == *gnu* ]]; then
+        compiler=gfort
+    else
+        "Unsure how to se the ./configure compiler from the selected compiler=$compiler"
+    fi
+fi
+
+echo
+printf "\e[7;49;94mModule information\e[0m\n"
+module purge
+# Is this strict enough in the sense that things might be changing?
+module load  $compiler_module  $mpi  ncarcompilers  netcdf  ncarenv || exit 4
+module list
+
+#-------------------------------------------------------
+# Python Env
+# Need option to install standardized python env.
+# Look in to pipenv some day and a required.txt file or similar
+# pip uninstall -y wrfhydropy
+# pip install wrfhydropy
+
+#-------------------------------------------------------
+## Candidates branch to tag the test directory and optionally update the reference.
+branch_name="$(git symbolic-ref HEAD 2>/dev/null)" || branch_name="$(git rev-parse --short HEAD)"
+branch_name=${branch_name##refs/heads/}
+printf "\e[7;49;94mTesting branch: $branch_name in $candidate_dir\e[0m\n"
+output_dir=/glade/scratch/`whoami`/take_test_$(basename $domain_dir)_${branch_name}
+
+#-------------------------------------------------------
+# "Update" the reference dir? This makes the branch match that of
+# the candidate and it fetches the NCAR:branch_name
+# There are two potential pitfalls here,
```

Comment:
I'm a little bit confused about why the default behavior here was chosen.  The first pitfall seems like it'll be pretty common and the automatic updating makes some assumptions for how repos are configured.

Reply:
It's just what was convenient for me... it was good for most of my cases. Should be good for most CI-ish cases. But I'm open to better ideas

Reply:
I think ultimately this should be an overridable default. But ok as a starting point.

Reply:
Agreed.